### PR TITLE
[5.1] SQL Server: Accept Table-Valued Function as table name

### LINKED ISSUE #11493: [5.1] Eloquent with sqlsrv connection can't use Table-valued Functions as Table
When creating a model that extends Eloquent and has a connection using the sqlsrv driver, if you set the table property to be a MSSQL table-value function it can't run any queries because it wraps the function in square brackets and crahes with "SQLSTATE[HY000]: General error: 208 General SQL Server error: Check messages from the SQL Server"

Meaning if set the table property to 

myTableValueFunction()

the query that runs later uses

...from [myTableValueFunction()]

If the query was instead 

...from \[myTableValueFunction]()

then it would work.  Even

...from myTableValueFunction()

would work.

### PR DESCRIPTION
**Pull request by rodrigopedra**
Closes #11493

### CODE REVIEW

File: src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php [hunk 1]
```diff
@@ -315,4 +315,22 @@ public function compileUpdate(Builder $query, $values)
 
         return trim("update {$table}{$joins} set $columns $where");
     }
+
+    /**
+     * Wrap a table in keyword identifiers.
+     *
+     * @param  string|\Illuminate\Database\Query\Expression  $table
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php]: We normally put scalars after objects in phpdoc.

Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php]: `\Illuminate\Database\Query\Expression|string`

**Comment by GrahamCampbell** [comment]
Aside from that minor point (which I suspect we might have inconsistencies already in the code base anyway), this looks good to me.

:+1:

File: src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php (hunk 1 above)
Comment (rodrigopedra) [inline src/Illuminate/Database/Query/Grammars/SqlServerGrammar.php]: I copied it from https://github.com/laravel/framework/blob/5.2/src/Illuminate/Database/Grammar.php#L27-L33

Want me to fix it there too?

**Comment by rodrigopedra** [comment]
Fixed the phpdoc, I will submit the `Illuminate\Database\Grammar` phpdoc fixes in another commit

**Comment by taylorotwell** [comment]
Thanks

**Pull request merged**